SandWorm/SandWormComponent.cs

```diff
@@ -138,209 +137,186 @@ protected override void SolveInstance(IGH_DataAccess DA)
             unitsMultiplier = Core.ConvertDrawingUnits(units); // Pick the correct multiplier based on the drawing units
             sensorElevation /= unitsMultiplier; // Standardise to mm to match sensor units
 
-            Stopwatch timer = Stopwatch.StartNew(); //debugging
+            Stopwatch timer = Stopwatch.StartNew(); // Setup timer used for debugging
+            output = new List<string>(); // For the debugging log lines
 
             if (this.kinectSensor == null)
             {
                 KinectController.AddRef();
                 this.kinectSensor = KinectController.sensor;
             }
-
-            if (this.kinectSensor != null)
+            if (this.kinectSensor == null)
+            { 
+                ShowComponentError("No Kinect sensor detected.");
```

Comment:
This condition is never true. Started SandWorm without Kinect connected and it evaluated to false, caught the exception from Line 153, though.

Reply:
Ah right, I just inverted the logic that was there previous, wasn't sure exactly how the checks work. Will remove that block.